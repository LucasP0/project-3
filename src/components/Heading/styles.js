import styled, { css } from 'styled-components';

const TitleSize = {
  xsmall: (theme) => css`
    font-size: ${theme.fonts.sizes.xsmall};
  `,
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.small};
  `,
  medium: (theme) => css`
  font-size: ${theme.fonts.sizes.medium};
  `,
  large: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  xlarge: (theme) => css`
   font-size: ${theme.fonts.sizes.xlarge};
  `,
  xxlarge: (theme) => css`
  font-size: ${theme.fonts.sizes.xxlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.huge};
  `,
  xhuge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${TitleSize[size](theme)};
  `}
`;
