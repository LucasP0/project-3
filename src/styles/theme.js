export const theme = {
  colors: {
    mainBg: 'red',
    secundaryBg: 'blue',
    rosinha: 'white',
  },
  fonts: {},
  spacings: {},
};
