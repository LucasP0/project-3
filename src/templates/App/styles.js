import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme, background }) => css`
  background: ${background};
  `}
`;
export const Lucas = styled.div`
  ${({ theme, background }) => css`
  background: ${background};
  `}
`;
