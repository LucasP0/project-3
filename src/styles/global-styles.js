import { createGlobalStyle, css } from 'styled-components';
import { theme } from './theme';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,900;1,900&family=Open+Sans:ital,wght@0,300;0,700;1,700&display=swap');

  *{
    margin:0px;
    padding: 0px;
    box-sizing: border-box;
  }
  html {
    font-size: 62.5%;
  }
  body {
    font-size: 1.6rem;
    font-family: 'Open Sans', sans-serif;
  }
  h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat', sans-serif;
  }
`;
