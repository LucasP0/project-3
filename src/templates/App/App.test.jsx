import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import Home from './index';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'HELLOOO WORD!!!',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: 'blue',
  });
  expect(headingContainer).toHaveStyleRule('background', 'blue');
});
