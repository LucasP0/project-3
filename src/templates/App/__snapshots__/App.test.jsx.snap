// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="sc-bcXHqe gMxvTB"
>
  <h1>
    HELLOOO WORD!!!
  </h1>
</div>
`;
